### PR TITLE
feat(git-changelog): multiple authors

### LINKED ISSUE #42: bug(git-changelog): Co-authored-by not recognized
In this documentation [为什么 Nólëbase 从 Creative Commons 协议的 BY-NC-SA 切换到了 BY-SA？ | Nólëbase](https://nolebase.ayaka.io/%E7%AC%94%E8%AE%B0/%F0%9F%93%9A%20%E4%B8%AA%E4%BA%BA%E7%9F%A5%E8%AF%86%E7%AE%A1%E7%90%86/%E4%B8%BA%E4%BB%80%E4%B9%88%20N%C3%B3l%C3%ABbase%20%E4%BB%8E%20Creative%20Commons%20%E5%8D%8F%E8%AE%AE%E7%9A%84%20BY-NC-SA%20%E5%88%87%E6%8D%A2%E5%88%B0%E4%BA%86%20BY-SA%EF%BC%9F.html) with commit at [9f4336d](https://github.com/nolebase/nolebase/commit/9f4336d12e4b7e6d0c2b566a81acfe1e4f2a5fde), the `Co-authored-by` attribute wasn't recognized by `vitepress-plugin-git-changelog`.

### PR DESCRIPTION
**Pull request by LittleSound**
close #42

### CODE REVIEW

File: packages/vitepress-plugin-git-changelog/src/client/components/Contributors.vue [hunk 1]
```diff
@@ -20,48 +20,66 @@ const options = inject(InjectionKey, {})
...
 const commits = useCommits(Changelog.commits, rawPath)
+const findRegisteredCreatorByName = (author_name: string) => options.mapContributors?.find(item => item.nameAliases && Array.isArray(item.nameAliases) && item.nameAliases.includes(author_name))
+const findRegisteredCreatorByEmail = (author_email: string) => options.mapContributors?.find(item => item.emailAliases && Array.isArray(item.emailAliases) && item.emailAliases.includes(author_email))
 
-const contributors = computed<ContributorInfo[]>(() => {
-  const map: Record<string, ContributorInfo> = {}
-  commits.value.forEach((c) => {
-    const targetCreatorByName = options.mapContributors?.find(item => item.nameAliases && Array.isArray(item.nameAliases) && item.nameAliases.includes(c.author_name))
-    const targetCreatorByEmail = options.mapContributors?.find(item => item.emailAliases && Array.isArray(item.emailAliases) && item.emailAliases.includes(c.author_email))
+const multipleAuthorsRegex = /^ *?Co-authored-by:(.*)(?:<|\(|\[|\{)(.*)(?:>|\)|\]|\}) *?$/mi
```
Comment (nekomeowww) [inline packages/vitepress-plugin-git-changelog/src/client/components/Contributors.vue:26]: - Should we add comments to explain what this Regexp does?
- What will happen on Safari when `?=`, `?<=`, `?!`, `?<!` is uses?
- Is `co-authored-by` an acceptable case too?

Comment (LittleSound) [reply packages/vitepress-plugin-git-changelog/src/client/components/Contributors.vue:28]: - I added some comments
- The syntax you listed is not in this expression.
- The `i` at the end of the regular expression means case insensitive.

File: packages/vitepress-plugin-git-changelog/src/vite/index.ts [hunk 2]
```diff
@@ -80,7 +80,7 @@ export function GitChangelog(options: {
         // get change log of each file
         // const raw = await git.raw(['diff-tree', '--no-commit-id', '--name-only', '-r', log.hash])
         const raw = await git.raw(['diff-tree', '--no-commit-id', '--name-status', '-r', '-M', log.hash])
-        delete log.body
+        // delete log.body
```
Comment (nekomeowww) [inline packages/vitepress-plugin-git-changelog/src/vite/index.ts:83]: Should we remove such commented code in this PR or add `// TODO:` for it to remove later?

Should we explain why we have such code left here?

**Review by nekomeowww — APPROVED** [review]

File: packages/vitepress-plugin-git-changelog/src/vite/index.ts (hunk 2 above)
Comment (LittleSound) [reply packages/vitepress-plugin-git-changelog/src/vite/index.ts:83]: We can create a new issue to discuss moving more work to build timen.

Comment (nekomeowww) [reply packages/vitepress-plugin-git-changelog/src/vite/index.ts:83]: Created https://github.com/nolebase/integrations/issues/45

**Pull request merged**